### PR TITLE
Use OpenIdSettings.Audiences as the default resources when no "resource" parameter is specified

### PR DESCRIPTION
**Pull request by PinpointTownes**
Fixes https://github.com/OrchardCMS/Orchard2/issues/870 plus a few other things (e.g the `Accept()` POST action didn't correctly check whether the application was allowed to use the specified flow).

### CODE REVIEW

File: src/OrchardCore.Modules/Orchard.OpenId/Controllers/AccessController.cs [hunk 1]
```diff
@@ -380,18 +343,16 @@ public IActionResult Deny()
                 });
             }
```
Comment (jersiovic) [inline src/OrchardCore.Modules/Orchard.OpenId/Controllers/AccessController.cs]: blank line ;)

File: src/OrchardCore.Modules/Orchard.OpenId/Controllers/AccessController.cs [hunk 2]
```diff
@@ -380,18 +343,16 @@ public IActionResult Deny()
...
-            if (_userManager.SupportsUserTwoFactor && await _userManager.GetTwoFactorEnabledAsync(user))
+            else if (result.RequiresTwoFactor)
             {
                 return BadRequest(new OpenIdConnectResponse
                 {
                     Error = OpenIdConnectConstants.Errors.InvalidGrant,
-                    ErrorDescription = T["The specified user is not allowed to sign in."]
+                    ErrorDescription = T["The specified user is not allowed to sign in using the password method."]
                 });
             }
```
Comment (jersiovic) [inline src/OrchardCore.Modules/Orchard.OpenId/Controllers/AccessController.cs]: same

Comment (PinpointTownes) [inline src/OrchardCore.Modules/Orchard.OpenId/Controllers/AccessController.cs]: Fixed (I tend to always add blank lines between if/else if/else clauses to lighten my code :sweat_smile: )

**Comment by PinpointTownes** [comment]
Merged. @jersiovic thanks for the review :+1:

**Pull request merged**